sigal/gallery.py

```diff
@@ -101,8 +102,10 @@ def big(self):
                 return self.filename
             orig_path = join(s['destination'], self.path, s['orig_dir'])
             check_or_create_dir(orig_path)
-            copy(self.src_path, join(orig_path, self.src_filename),
-                 symlink=s['orig_link'])
+            big_path = join(orig_path, self.src_filename)
+            if not os.path.isfile(big_path):
+                copy(self.src_path, big_path,
+                     symlink=s['orig_link'])
```

Comment:
Could be a good idea. But the main problem I had with that is that for example in the galleria template, snippets like `{%- if media.big %}<a href='{{ media.big }}'>Full size</a>{% endif %}` calls 2 times this function. Another test on `media.big` is done later so the big image is copied 4 times for nothing.

What do you think of the idea of implemeting a test with md5 hash of the file to avoid the copy and still be sure that an overwritten big image will still be correctly copied ?